### ISSUE TITLE
Allow controlling Wayland decoration colors separately from terminal colors in config

### ISSUE THREAD

**Issue by Qyriad** [alacritty/alacritty#4077]
After #3455, the Wayland decorations are based on `colors.primary.background` and `colors.primary.foreground`, but I'd like the decoration colors to be different from my terminal background and foreground colors. Could they be pulled out into separate configuration options? Possibly with `colors.primary.background` and `colors.primary.foreground` set as the default?

**Comment by chrisduerr** [comment]
We've intentionally decided against this to keep the number of configuration options down a bit. If users dislike the client side decorations, then there's always the option to just use server side decorations.

Is there a reason why server side decorations do not work for you? I'd imagine those will be nicer than Alacritty's CSDs anyways.

**Comment by Qyriad** [comment]
I…must admit that I didn't realize that was an option, and now I feel a little silly, because I still don't know how I would go about using server side decorations for Alacritty on Gnome (…if the answer isn't "you can't" on Gnome). If I can get server-side decorations working that would probably be fine.

**Comment by chrisduerr** [comment]
> if the answer isn't "you can't" on Gnome

It is. You've got gnome to thank for that. But that's something that should be picked up with gnome.

I'm curious what @kchibisov thinks about this, since he's more into Wayland than I am. But I'd say since there is a Wayland spec that solves this issue and adding an entire config block just for a single display server is a lot, I find it difficult to justify this.

**Comment by kchibisov** [comment]
>I'm curious what @kchibisov thinks about this, since he's more into Wayland than I am. But I'd say since there is a Wayland spec that solves this issue and adding an entire config block just for a single display server is a lot, I find it difficult to justify this.

In general, if we want to allow some configuration for them, I'd suggest some side channel for that, like env variable or some other thing. I'm not even if I want to keep such theming inside of alacritty directly tbqh, especially knowing that you have not just 2 options for text and background, but quite a lot of them (buttons, different focus states, etc).

Maybe in general having some common way to parse some theme config(simple ini file) right in sctk(behind some feature) could do something that makes sense, but I can't say whether such thing could even be accepted in it.

**Comment by chrisduerr** [comment]
Having a theme config file in SCTK would be slightly strange imho, but I suppose makes some sense since other UI toolkits do the same (though generally they're more involved in the looks of the application).

But this would mean that you're also against doing this in Alacritty, right? Then I'd close this as wontfix with the possibility that it might some day be resolved through other means.

**Comment by kchibisov** [comment]
The alacritty approach I'm suggesting is to have some env variable and maybe pass only a few options via it? idk.

**Comment by chrisduerr** [comment]
Having some random environment variable to control configuration options is not something I'm interested in. At that point I'd rather users that *really* need it just compile a small patch from source.

I'm going to close this as wontfix, simply because Alacritty should strive to be simple and getting all caught up in the different decorations and their color is just a bit too much of a mess. Supporting CSDs alone is already quite a lot of work that thankfully someone else does for us (unfortunately with no way around it on Wayland). But they should be seen as a last resort, if they don't work to your liking just use SSDs. Let's keep things simple.
